Posts/2021/02/Get-Services, PowerShell 7 VS Windows PowerShell 5.1

```diff
@@ -0,0 +1,150 @@
+ServiceName                    Property
+ServicesDependedOn             Property
+ServiceType                    Property
+Site                           Property
+StartType                      Property
+StartupType                    Property
+Status                         Property
+UserName                       Property
+ToString                   ScriptMethod
+```
+
+```powershell
+**Windows Powershell 5.1**
+Name                         MemberType
+----                         ----------
+Name                      AliasProperty
+RequiredServices          AliasProperty
+Disposed                          Event
+Close                            Method
+Continue                         Method
+CreateObjRef                     Method
+Dispose                          Method
+Equals                           Method
+ExecuteCommand                   Method
+GetHashCode                      Method
+GetLifetimeService               Method
+GetType                          Method
+InitializeLifetimeService        Method
+Pause                            Method
+Refresh                          Method
+Start                            Method
+Stop                             Method
+WaitForStatus                    Method
+CanPauseAndContinue            Property
+CanShutdown                    Property
+CanStop                        Property
+Container                      Property
+DependentServices              Property
+DisplayName                    Property
+MachineName                    Property
+ServiceHandle                  Property
+ServiceName                    Property
+ServicesDependedOn             Property
+ServiceType                    Property
+Site                           Property
+StartType                      Property
+Status                         Property
+ToString                   ScriptMethod
+```
+
+## PowerShell 7
+
+The big difference is in the Property, now in Powershell 7, it's possible to read some additional properties that were not available in Windows PowerShell 5.1, such as **UserName, BinaryPathName, StartType.** So let's see how to read the **Username**, **BinaryPathName,** and **StartType** using PowerShell 7
+
+```powershell
+PS 7> Get-Service workstation | select  Username,Starttype,BinaryPathName
+```
+
+The output is a clear, with all the required result with one-liner native command.
+
+```powershell
+UserName                    StartType BinaryPathName
+--------                    --------- --------------
+NT AUTHORITY\NetworkService Automatic C:\WINDOWS\System32\svchost.exe -k NetworkService -p
+```
+
+## Windows Powershell 5.1
+
+For Windows PowerShell 5.1, the operation is not straight forward, as such in PowerShell 7, we need to use the `Get-CimInstance` and pass the required WQL query, so in Windows PowerShell 5.1 run the following command
+
+```powershell
+WPS 5.1> Get-CimInstance -Query 'select * from Win32_Service where caption like "Workstation"' | select StartName,StartMode,PathName
```

Comment:
to simify things, why not just 

```powershell
Get-Ciminstance -Class Win32_Service | Where-Object Name -eq LanManWorkstation |
  Select-Object -Property StartName, StartMode, PathName
```

Reply:
Yes, sure, we can add it as a this or that but will make it like, `Where-Object Caption -eq Workstation` this won't confuse the reader as the **LanManWorkStation** never was mentioned in the post and the post is only discussing a service named **Workstation**
And to be honest I chose to write a query just to show that things can get a bit complicated in WMI :)